lib/replicator.js

```diff
@@ -1,851 +1,1472 @@
-const Protocol = require('./protocol')
-const RemoteBitfield = require('./remote-bitfield')
-const RandomIterator = require('random-array-iterator')
 const b4a = require('b4a')
+const sodium = require('sodium-universal')
+const safetyCatch = require('safety-catch')
+const RandomIterator = require('random-array-iterator')
+const RemoteBitfield = require('./remote-bitfield')
+const m = require('./messages')
 
-const PKG = require('../package.json')
-const USER_AGENT = PKG.name + '/' + PKG.version + '@nodejs'
+const DEFAULT_MAX_INFLIGHT = 32
 
-class RemoteState {
-  constructor (core) {
-    this.receivedInfo = false
-    this.inflight = 0
-    this.maxInflight = 16
-    this.bitfield = new RemoteBitfield()
-    this.length = 0
-    this.fork = 0
-    this.uploading = true
-    this.downloading = true
-  }
-}
+const NS = b4a.allocUnsafe(32)
```

Comment:
Should this use `crypto.namespace()`?

Reply:
Ya I need to update crypto.namespace (chris reviewed it and made it simpler). I'll update